src/json.rs

```diff
@@ -0,0 +1,55 @@
+//! structured data store support for `sled` db
+//! # Examples
+//!
+//! ```
+//! use sled;
+//! use serde::{Deserialize, Serialize};
+
+//! #[derive(Serialize, Deserialize, Debug)]
+//! struct Info {
+//!     name: String,
+//!     age: u8,
+//! }
+
+//! let linus_walker = Info { name: "Linus Walker".into(), age: 14 };
+//! let db = sled::open("db").unwrap();
+//! db.insert_json("linus_walker", &linus_walker).unwrap();
+//! let resul = db.get_structued::<Info>(&"linus_walker".to_string()).unwrap();
+//! println!("{resul:?}");
+//!}
+//! ```
+
+use crate::{Db, IVec, Result};
+use serde::{de::Deserialize, ser::Serialize};
+
+impl Db {
+    /// function implementation for `Db` to store structured data as json in db
+    pub fn insert_json<K: AsRef<[u8]>, V: ?Sized + Serialize>(
+        &self,
+        key: K,
+        value: &V,
+    ) -> Result<Option<IVec>> {
+        let value = serde_json::to_string(value).unwrap();
```

Comment:
Would be best not to unwrapping as that would result in a panic if it fails to serialize (you could probably use io::Error or just one of the other errors in `result::Error`). Also, wouldnt it be better to use `serde_json::to_vec` instead?